extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,479 @@
+    boolean response = emailService.isValid( emailConfigOriginal );
+    assertEquals( true, response );
+  }
+
+  @Test
+  public void testInvalidEmailConfig() {
+
+    final EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "", "", "smtp.com", 36,
+                    EmailConstants.PROTOCOL_SMTP, true, "", "", false, true );
+
+    boolean response = emailService.isValid( emailConfigOriginal );
+    assertEquals( false, response );
+  }
+
+  @Test
+  public void testSendEmailTestNoAuth() throws Exception {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( false, false, "test@pentaho.com", "Pentaho Scheduler", "", 25,
+                    EmailConstants.PROTOCOL_SMTP, true, "", "", true, true );
+    assertEquals( EmailService.TEST_EMAIL_SUCCESS, emailService.sendEmailTest( emailConfigOriginal ) );
+  }
+
+  @Test
+  public void testSendEmailGraphApi() throws Exception {
+    MockWebServer server = new MockWebServer();
+    String filename = "EmailService/EmailAuthenticationResponse1.json";
+    String content = readTestFile( filename );
+
+    server.enqueue( new MockResponse()
+            .setResponseCode( HttpStatus.SC_OK )
+            .setBody( content )
+    );
+
+    String urlPathToken = "/common/oauth2/v2.0/token";
+    String mockUrlFulToken = server.url( urlPathToken ).toString();
+
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "", 25,
+                    EmailConstants.PROTOCOL_GRAPH_API, true, "test", "", true, true );
+    emailConfigOriginal.setAuthMechanism( EmailConstants.AUTH_TYPE_XOAUTH2 );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setTokenUrl( mockUrlFulToken  );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( EmailConstants.GRANT_TYPE_REFRESH_TOKEN );
+    emailConfigOriginal.setRefreshToken( "refreshToken" );
+    Properties emailProperties = new Properties();
+    Session session = Session.getInstance( emailProperties );
+    MimeMessage msg = new MimeMessage( session );
+    msg.setFrom( new InternetAddress( emailConfigOriginal.getDefaultFrom(), emailConfigOriginal.getFromName() ) );
+    msg.setRecipients( Message.RecipientType.TO, InternetAddress.parse( emailConfigOriginal.getDefaultFrom() ) );
+    msg.setSubject( "SUBJECT" );
+    msg.setText( "EmailService.MESSAGE" );
+    msg.setHeader( "X-Mailer", "smtpsend" );
+    msg.setSentDate( new Date() );
+    try {
+      emailService.sendEmail( emailConfigOriginal, session, msg );
+    } catch ( EmailServiceException e ) {
+      assertTrue( e.getMessage().contains( "401 Unauthorized" ) );
+    }
+  }
+
+  @Test
+  public void testSendEmailSmtp() throws Exception {
+    MockWebServer server = new MockWebServer();
```

Comment:
Do you have any unit tests that verifies the SMTP process works?

This might have be addressed in the refactor. Once the code is changed to this: https://github.com/pentaho/pentaho-platform/pull/5185/files#r1091375072

Then you can test via MockMail